src/dmd/dsymbolsem.d

```diff
@@ -2108,7 +2108,117 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
         //printf("UserAttributeDeclaration::semantic() %p\n", this);
         if (uad.decl && !uad._scope)
             uad.Dsymbol.setScope(sc); // for function local symbols
-        return attribSemantic(uad);
+        attribSemantic(uad);
+        if (uad.atts)
+        {
+            // detect the special attribute @__gnu_abi_tag(...)
```

Comment:
This looks like a magic name. I would prefer to implement this as a true `UDA` which can use the fully qualified name to resolve any conflicts. This would avoid the need for the ugly double underscore prefix. You can have a look at how the `@selector` attribute is implemented.

Reply:
I'll take a look.

Reply:
Yes, that is better approach. I switched too it. I didn't make it part to druntime yet though.
I also ended up making an exception for namespaces from the UDA inheritance rules to stop members from inheriting this special UDA from namespaces, this how it behaves in C++.

Reply:
Here is the `core.attribute` addition https://github.com/dlang/druntime/pull/2639. Once it gets in I'll activate the module checks.